Beta v7.0.-1

MichaIng opened the pull request:
### Beta v7.0.-1
_(2021-02-18)_

#### Changes
- DietPi-Update | A new update check system has been implemented, which checks for a minimum required Debian and DietPi version and migrates systems with either too old Debian version or too old DietPi version to a different Git branch automatically. This especially smoothens the migration of our code base to newer Debian versions and the alternative branch can be used to inform and support Debian distribution upgrades and to upgrade the DietPi code to an intermediate version from where the system can be migrated back to the regular stable/master branch. Since this change required a new repository version file, we took the chance and incremented the DietPi core version to v7.0. This however has no further meaning, all previously supported DietPi and Debian versions remain supported and the change enables us to not require fresh image installs for major upgrades, like the upgrade from v159 to v6.0 did a few years ago.
- DietPi-Backup | Support for XFS and ZFS target file system types has been added, which fully support the required symlink and POSIX permissions capabilities.
- DietPi-Software | The mandatory reboot after installs has been removed. Installed services, which are not controlled by DietPi-Services, but would start automatically on reboot, are now started at the end of installs instead. A manual reboot is still a good idea, but strictly required only in rare cases. Many thanks to @Games-Crack for doing this suggesting: https://github.com/MichaIng/DietPi/issues/4032
- DietPi-Software | Installs do not imply all APT package upgrades anymore. While we recommend to keep all APT packages upgraded regularly, the new daily APT check and info within the DietPi-Banner, helps to keep you informed to do the best decision youself whether and when to apply which package upgrade. On first run installs the full upgrade is however kept, to assure each image starts in fully upgraded state, and packages which are required for the actual software choices you're installing, are upgraded as well, when installed already.
- DietPi-Software | Uninstalls do not stop other services anymore. E.g. your webserver or media streaming server will stay active while you uninstall that other software that you don't require anymore. Since uninstalls do not require much RAM or CPU resources, this is perfectly fine. Many thanks to @mrgreaper giving the hint: https://github.com/MichaIng/DietPi/issues/4116
- DietPi-Software | Unbound: On install in combination with Pi-hole, no additional configuration file will be created anymore but the adjusted interface binding and port will be applied to "/etc/unbound/unbound.conf.d/dietpi.conf". Declaring "interface" in two configuration files do not override each other but lead to two concurrent bindings, which is not intended. The two files, if present, will be merged as well on DietPi update. It is hence intended that admins change "dietpi.conf" directly, if required, and this file won't be overwritten on reinstalls to preserve local changes. Additionally, on new installs, the configuration file will be better sorted and contain comments to explain their purpose.
- DietPi-Software | Unbound: On new installs, by default access is now granted to all private IPv4 and IPv6 address ranges instead of to the 192.168.0.0/16 subnet only, which includes VPNs, containers and cases of multiple local networks the server is attached to.
- DietPi-Software | Unbound: A monthly cron job is now created to keep the root hints file updated. Many thanks to @APraxx for doing this suggestion: https://github.com/MichaIng/DietPi/issues/4077

#### Interface Changes
- DietPi-Banner | Running the script without input argument will now open the menu instead of printing the banner. The console alias has been adjusted accordingly, so that it is now possible to run "dietpi-banner 0" and "dietpi-banner 1" from console to have full and short banner versions printed. The "dietpi-banner" console command remains opening the menu.

#### New Software
- DietPi-Software | Docker Compose: A tool to define and run multi-container Docker applications can now be installed through our software selection. Docker will be pulled in as dependency automatically.
- DietPi-Software | Box86: An x86 wrapper/emulator for ARMv7 systems is now available for install. Thanks to it's ability to wrap ARMv7 shared system libraries to be used with i386 binaries, often no additional libraries need to be installed. Thanks to binfmt, it will be invoked automatically when an i386 binary is executed.
- DietPi-Software | Steam: By automatically pulling in Box86 as dependency, Steam can now be installed on ARMv7 boards. It won't run perfectly stable yet and has limited features and game support, but we're optimistic that further improvements will address this in the future. Check out our documentation about it: https://dietpi.com/docs/software/gaming/#steam
- DietPi-Software | mjpg-streamer: Lightweight multi-source and multi-output plugin JPEG streamer, now available for install. By default your attached camera capture will be streamed to a custom HTTP port. When installed along with OctoPrint, both will be automatically configured to work together. The Raspberry Pi camera module support is enabled by default, when installed on Raspberry Pi.

#### Fixes
- DietPi-Set_swapfile | Resolve an issue where "zram"/"zram0" dietpi.txt path entries were dropped, when running the script without input arguments. This especially broke applying zram-swap on first boot. Many thanks to @Dr0bac for reporting this issue: https://github.com/MichaIng/DietPi/issues/4002
- DietPi-Software | Bitwarden_RS: Resolved an issue where the self-signed TLS certificate could not be imported on iOS. To apply this fix to an existing instance, the configuration file "/mnt/dietpi_userdata/bitwarden_rs/bitwarden_rs.env" needs to be removed or moved to a different location, so "dietpi-software reinstall 183" will re-create the configuration and TLS certificate.
- DietPi-Software | Unbound: Resolved an issue where during install in combination with Pi-hole the service restart could have failed. Many thanks to @Ernstian for reporting this issue: https://github.com/MichaIng/DietPi/issues/2409#issuecomment-739154892
- DietPi-Software | Unbound: Resolved an issue where the service start failed if the host system had a local IP address outside of the 192.168.0.0/16 subnet. Many thanks to @faxesystem for reporting this issue: https://github.com/MichaIng/DietPi/issues/2409#issuecomment-749174984
- DietPi-Software | ReadyMedia: Resolved an issue where the media library was not rescanned on service start. Many thanks to `@AdamFarnsworth0` for reporting this issue: https://twitter.com/AdamFarnsworth0/status/1347977813635305475
- DietPi-Software | WiFi Hotspot: Resolved an issue where the install on Armbian-based images with RTL8188C* WiFi chip failed. Many thanks to `@smogan71` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?t=8523
- DietPi-Software | Medusa: This software option has been disabled on Stretch systems, since it now requires Python >=3.6, which is not available in the Debian Stretch repository. If you run Medusa on a Stretch system, it will continue to work, but updating will either not be possible or break it. Many thanks to @aermak for reporting this issue: https://github.com/MichaIng/DietPi/issues/3991
- DietPi-Software | WiringPi: Resolved an issue where the install failed, if the /usr/local/bin directory was not present. Many thanks to `@bruz` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?t=8609
- DietPi-Software | PaperMC: Resolved an issue where the install failed due to changed download URLs and stabilised service start and config creation by setting the Java heap size and allowing more time for the startup on smaller SBCs. Many thanks to `@omavoss` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?p=30191#p30191
- DietPi-Software | OpenTyrian: The install option has been disabled on x86_64, since the Debian package comes with a different file structure and it has been disabled on 64-bit RPi systems, since the binary has been compiled for armhf.
- DietPi-Software | Domoticz: Resolved an issue where saving custom scripts and starting with a template did not work. Many thanks to `@tec13` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?t=8627
- DietPi-Software | Resolved an issue where for ruTorrent, Koel and Bitwarden_RS the automatic newest version detection failed and instead a possibly older fallback was used. Many thanks to @kelvmod for reporting this issue: https://github.com/MichaIng/DietPi/issues/4105
- DietPi-Software | LXQt: Resolved visual issues with our default configuration of Debian Buster, drastically simplified and cleaned up the files we ship.

MichaIng committed:
Release v6.34 (#3982)

MichaIng committed:
v6.35 (#3987)

+ META | Backwards merge

MichaIng committed:
v6.35 (#3988)

+ Prepare v6.35 development cycle

MichaIng committed:
v6.35

+ DietPi-Survey_report | Syntax fix

MichaIng committed:
v6.35

+ DietPi-Patch | Allow updating to v6.35 without endless loop xD

MichaIng committed:
v6.35

+ DietPi-PREP | Fix step headers, so that script name is on top of the HR and the step count and text below after "Step: "

MichaIng committed:
v6.35

+ DietPi-Software | Rename "Minio" => "MinIO": https://min.io/

MichaIng committed:
v6.35

+ DietPi-LetsEncrypt | Renamed "Minio" => "MinIO": https://min.io/

MichaIng committed:
v6.35

+ DietPi-Survey_report | Renamed "Minio" => "MinIO": https://min.io/

MichaIng committed:
v6.35

+ CHANGELOG | Renamed "Minio" => "MinIO": https://min.io/

MichaIng committed:
v6.35

+ DietPi-Patch | Renamed "Minio" => "MinIO": https://min.io/
+ DietPi-Patch | Dropbear: Do not create DSS host key anymore. It is pretty outdated and should not be used anymore. Since Bullseye it is not shipped by default, but Ed25519 instead.

MichaIng committed:
v6.35

+ DietPi-PREP | Remove obsolete zsh configs from user profiles

nonnorm committed:
v6.35 (#3995)

+ DietPi-Software | LXDE/LXQt: Allow configs to be pulled from custom GitHub repository owner instead of custom branch only

MichaIng committed:
v6.35

+ DietPi-Login | Create the "dietpi-nordvpn" alias unconditionally. The script handles a missing install gracefully and offers to do the install right away. It is especially unpleasing to not have the alias available after install, since it was needed to setup NordVPN server and credentials first, to have the service file created: https://github.com/MichaIng/DietPi/issues/3998

MichaIng committed:
v6.34

+ DietPi-Software | Bitwarden_RS: Add CA flag to self-signed certificate, required for iOS to allow import into trusted CA store: https://github.com/MichaIng/DietPi/issues/3724#issuecomment-750883187

MichaIng committed:
v6.35

+ CHANGELOG | Bitwarden_RS: Resolved an issue where the self-signed TLS certificate could not be imported on iOS

MichaIng committed:
v6.35

+ DietPi-Set_swapfile | Resolve an issue where "zram"/"zram0" dietpi.txt path entries were dropped, when running the script without input arguments. This especially broke applying zram-swap on first boot.

MichaIng committed:
v6.35

+ CHANGELOG | DietPi-Set_swapfile: Resolve an issue where "zram"/"zram0" dietpi.txt path entries were dropped, when running the script without input arguments.

Joulinar committed:
v6.35 (#4006)

+ DietPi-Software | Portainer: Ensure that post- and pre-v2.0 image versions of Portainer are detected during installation or removal

MichaIng committed:
v6.35

+ DietPi-Patch | Do not create dietpi-usb-storage_quirks.conf, as we do not statically apply USB quicks anymore
+ DietPi-Patch | Do not apply obsolete "ui" component on RPi Buster systems

MichaIng committed:
v6.35

+ DietPi-Services | Stop controlling obsolete CloudPrint but start controlling PHP8.0-FPM. Start HTPC Manager later, as it is a web UI frontend for various other software titles

MichaIng committed:
v6.35

+ DietPi-Software | Pi-hole: Remove blocking page support, which will be removed from Pi-hole soon. It's implementation relies on fragile 404 handling, which means that actual 404 handling does not work. It means that every blocked request will instead be fully answered by the Pi-hole webserver, which means a massive overhead for both server and client. It cannot work on HTTPS resources since the certificate cannot match the domain which means that a shrinking minor part of the web is only supported anyway. Finally ads often open in small frames rather than full screen popups, which means that the blocking page is often practically not usable. Instead we revert to the Pi-hole developers recommended NULL blocking mode, which means that most clients won't even attempt to do any request to this domain, hence least overheads for clients, most time, battery and traffic savings.

MichaIng committed:
v6.35

+ DietPi-Software | Pi-hole: Remove blocking page support which will be removed from Pi-hole

MichaIng committed:
v6.35

+ DietPi-Software | Pi-hole: Remove blocking page support which will be removed from Pi-hole

MichaIng committed:
v6.35

+ DietPi-Software | Pi-hole: Remove blocking page support which will be removed from Pi-hole

MichaIng committed:
v6.35

+ DietPi-Software | Pi-hole: Tiny

MichaIng committed:
v6.35

+ DietPi-PREP | Make it clear that Odroid C4 and HC4 are cross-compatible

MichaIng committed:
v6.35

+ DietPi-Obtain_HW_model | Make it clear by name that Odroid C4 and HC4 are cross-compatible

MichaIng committed:
v6.35

+ DietPi-Software | Unbound: Add default config file as GitHub download

MichaIng committed:
v6.35

+ DietPi-Software | Unbound: Pre-configure Unbound before installing it to avoid port binding conflicts right from the start
+ DietPi-Software | Unbound: Install it before installing Pi-hole so that it can be used as upstream DNS directly
+ DietPi-Software | Unbound: Use port 5335 by default when configuring it for Pi-hole since port 5353 is used for mDNS (Multicast DNS)
+ DietPi-Software | Unbound: Whenever changing the Unbound port, assure that there is another upstream DNS server configured in /etc/resolv.conf
+ DietPi-Software | Unbound: Do not overwrite an existing configuration file. Config overrides do not in cases where the setting is allowed multiple times, hence in case of interface (IP bindings) private and allowed IP ranges, users might need to configure our base config, which must be preserved.
+ DietPi-Software | Pi-hole: When changing the upstream DNS, apply it directly via dnsmasq config and restart FTL to immediately use it.
+ DietPi-Software | Spotify Connect Web: Enhance uninstall

MichaIng committed:
v6.35

+ DietPi-Software | Unbound: Syntax

MichaIng committed:
Update unbound branch to current dev (#4029)

MichaIng committed:
v6.35

+ DietPi-Software | Unbound: Sort and comment default config file

MichaIng committed:
v6.35

+ DietPi-Survey_report | Uploads not updated since first half of 2020 have been achieved

MichaIng committed:
v6.35

+ DietPi-Services | Remove dnsmasq completely. We show its status as being part of Pi-hole, but since v4 with FTLDNS, that is not the case anymore.

MichaIng committed:
v6.35

+ DietPi-Software | Unbound: Rephrase DNS port comment

MichaIng committed:
v6.35

+ DietPi-Patch | Unbound: Merge dietpi-pihole.conf into dietpi.conf

MichaIng committed:
v6.35

+ CHANGELOG | Unbound: Add fixes and changes

MichaIng committed:
v6.35

+ DietPi-Software | Unbound: Add access control for IPv6, since we set "do-ip6: yes", if IPv6 is enabled, which does not only allow Unbound to connect via IPv6 to upstream/DNS root but allows clients to connect to it via IPv6 as well.

MichaIng committed:
v6.35

+ CHANGELOG | Unbound: On new installs, by default access is now granted to all private IPv4 and IPv6 address ranges

MichaIng committed:
v6.35 (#4022)

+ DietPi-Software | Unbound: Pre-configure Unbound before installing it to avoid port binding conflicts right from the start
+ DietPi-Software | Unbound: Install it before installing Pi-hole so that it can be used as upstream DNS directly
+ DietPi-Software | Unbound: Use port 5335 by default when configuring it for Pi-hole since port 5353 is used for mDNS (Multicast DNS)
+ DietPi-Software | Unbound: Whenever changing the Unbound port, assure that there is another upstream DNS server configured in /etc/resolv.conf
+ DietPi-Software | Unbound: Do no create an override config for Pi-hole, as those do not work in cases where the setting is allowed multiple times/stacking, e.g. interface bindings, access control or private IP ranges. Alter the main config file instead.
+ DietPi-Software | Unbound: Do not overwrite an existing configuration file, since users need to alter it at least for settings that are allowed multiple times. Customisation must hence be preserved.
+ DietPi-Software | Unbound: Add access control for IPv6, since we set "do-ip6: yes", if IPv6 is enabled, which does not only allow Unbound to connect via IPv6 to upstream/DNS root but allows clients to connect to it via IPv6 as well.
+ DietPi-Software | Unbound: Sort and comment default config file
+ DietPi-Software | Pi-hole: When changing the upstream DNS, apply it directly via dnsmasq config and restart FTL to immediately use it.
+ DietPi-Software | Spotify Connect Web: Enhance uninstall
+ DietPi-Patch | Unbound: Merge dietpi-pihole.conf into dietpi.conf

StephanStS committed:
v6.35 (#4042)

+ DietPi-Software | Change documentation links from phpBB to new MkDocs-based documentation pages

MichaIng committed:
v6.35

+ DietPi-PREP | Odroid C4/N2: Do kernel postinst steps manually, which depends on /proc/cpuinfo content, so allow correct install from within a container/chroot environment

MichaIng committed:
v6.35

+ DietPi-Imager | Minor visual main menu enhancements

MichaIng committed:
v6.35

+ DietPi-Software | ReadyMedia (MiniDLNA): Resolve an issue where the media database was not recreated on service restart as expected. Remove obsolete defaults config file and log directory, created on service start at package install. When starting the daemon in forground "-S", no log file is created but logging is done to STDOUT => "journalctl -u minidlna". Use G_EXEC error handler where possible.

MichaIng committed:
v6.35

+ DietPi-Software | MiniDLNA: Remove obsolete log_dir setting, since it is not used in combination with "-S" foreground flag with systemd and "/var/log" is the default anyway. Add info about how to access logs to avoid confusion.

MichaIng committed:
Update CHANGELOG.txt

MichaIng committed:
v6.35

+ DietPi-Patch | ReadyMedia: Fix database rebuild on service start

MichaIng committed:
v6.35

+ DietPi-Software | Jellyfin: Disable install option for ARMv6 since it cannot work on this architecture as of .NET limitation. The "raspbian" distro in the repository exists, but is contains the same packages as the "debian" distro (symlink, rewrite), to allow using "ID" from /etc/os-release on Raspbian, but it does not contain Raspbian-like armv6hf binaries.

MichaIng committed:
v6.35

+ DietPi-Patch | Bring forward the haveged upgrade on ARMv7/8 as a new version with further fixes is available on Bullseye

MichaIng committed:
v6.35

+ DietPi-Set_hardware | I2C: Do not error out when /boot/dietpi/.installed does not (yet) exist. This can be the case when on first boot dietpi-software prompt users enter dietpi-config and enable I2C from there. This means that the software ID will not be shown as installed, but it's a very minor visual issue only and cannot be easily solved as long as we do not ship /boot/dietpi/.installed with fresh images, which is planned.

MichaIng committed:
v6.35

+ DietPi-Software | WiFi Hotspot: Typo

MichaIng committed:
v6.35

+ CHANGELOG | WiFi Hotspot: Resolved an issue where the install on Armbian-based images with RTL8188C* WiFi chip failed

MichaIng committed:
v6.35

+ DietPi-Patch | Replace wget with curl

MichaIng committed:
v6.35

+ DietPi-Set_software | Replace wget with curl

MichaIng committed:
Align survey page style with main website

MichaIng committed:
v6.35

+ DietPi-Software | Minor

MichaIng committed:
v6.35

+ DietPi-Imager | Remove child mounts (e.g. boot) automatically after finishing the rootfs review mount.

MichaIng committed:
v6.35

+ DietPi-Survey_report | Make tables scrollable on small screens instead making the page too large to read it without (auto-)zoom

MichaIng committed:
v6.35

+ DietPi-PREP | Remove console auto-login, present on some Armbian images: https://github.com/MichaIng/DietPi/issues/1443#issuecomment-762773403

MichaIng committed:
v6.35

+ DietPi-PREP | Remove auto-login override for serial console as well from Armbian-based images. Remove all console overrides and none of them is standard of required!

MichaIng committed:
v6.35

+ DietPi-Banner | @Camry2731 is our new Patreon Legend, many thanks for your support!
+ DietPi-Banner | Remove obsolete short image update notification, the long form has been removed with v6.34 already.

MichaIng committed:
v6.35

+ DietPi-Software | Amiberry: libegl1 is installed now when enabling RPi KMS driver, hence it doesn't need to be installed explicitly as Amiberry dependency anymore

MichaIng committed:
v6.35

+ DietPi-Set_hardware | RPi: Do not attempt to start hciuart service directly when enabling Bluetooth. A reboot is required to have the device enabled, otherwise the service start will fail.

MichaIng committed:
v6.35

+ DietPi-Config | Enable Bluetooth options on virtual machines. One can e.g. easily use a physical USB Bluetooth adapter from within the VM. Also for testing the options and scripts, this is good.
+ DietPi-Config | Do not force fake-hwclock to lower the system time on load or store timestamps even if those are lower than it's on release date. The chance that the initial system clock is in fact in the future is likely lower than the chance that this messes with other time sync daemons that updated the system clock prior to fake-hwclock, e.g. systemd-timesyncd which has the same feature implemented, when being enabled (dietpi-config time sync daemon + drift mode).
+ DietPi-Config | On RPi, when enabling onboard Bluetooth, a reboot is required to have the device tree changes taking effect.

MichaIng committed:
v6.35

+ DietPi-PREP | Do not force fake-hwclock to lower the system time on load or store timestamps even if those are lower than it's on release date. The chance that the initial system clock is in fact in the future is likely lower than the chance that this messes with other time sync daemons that update the system clock prior to fake-hwclock, e.g. systemd-timesyncd which has the same feature implemented, when being enabled (dietpi-config time sync daemon + drift mode).

MichaIng committed:
v6.35

+ DietPi-Set_hardware | RPi: Do not add device tree overlays or parameters at runtime. We already know that this does not work in most cases but a reboot is required.
+ DietPi-Set_hardware | RPi4: The VC4 full KMS driver on RPi 4 additionally blocks card 1 with additional sound devices for the second HDMI port. Increment the target sound card hence additionally by 1 if this driver is enabled.

MichaIng committed:
v6.35

+ DietPi-Config | When selecting a not auto-detected sound card, a reboot is required to apply device tree changes

Joulinar committed:
v6.35 (#4020)

+ DietPi-Software | Added Docker-Compose software option with ID 134, with dependencies Docker and Python 3 pip
+ DietPi-Software | Python 3 pip: Add pipweels as additional pip repository for ARMv6 and ARMv7

MichaIng committed:
v6.35

+ DietPi-Globals | G_EXEC: Use a backslash-escaped command string whenever it is printed through G_DIETPI-NOTIFY to avoid expansion for escape sequences like colour codes and especially newlines

MichaIng committed:
v6.35

+ DietPi-Set_hardware | soundcard:  Avoid error messages when no sound card is currently registered. Apply G_EXEC and print a general message when resetting and applying new chosen settings so separate the additional output.
+ DietPi-Set_hardware | soundcard: Prefer /proc/asound/cards over "aplay -l" to gather sound card information

MichaIng committed:
v6.35

+ DietPi-Patch | Inform users of pre-v6.14 systems interactively about the global software password encryption and removal from dietpi.txt. This was done in dietpi-software previously but fits much better right into the update process where we inform about such changes and assume hence an interactive call anyway. Instead of applying the informational sentence as string to dietpi.txt, comment the setting as well so that it is not used accidentally a new input password, if that dietpi.txt is copied unchanged to a new DietPi system.

MichaIng committed:
v6.35

+ DietPi-Software | Do not print info about password encryption anymore. This is not done as part of dietpi-update patches and on new images as part of dietpi-set_software call as part of first run setup.
+ DietPi-Software | When password fallback is used, always re-encrypt it and inform users how to change it.
+ DietPi-Software | When password decryption fails with an exit code, assure that the string is empty to allow fallback to the default, so that no unexpected string can be applied.
+ DietPi-Software | When encrypting the password, instead of applying the informational sentence as string to the dietpi.txt, comment the setting as well so that it is not used accidentally as new input password, e.g. if that dietpi.txt is copied unchanged to a new DietPi system. of the encrypted file got removed.

MichaIng committed:
v6.35

+ dietpi.txt | Add info that the global software password will be removed from the config file on first dietpi-software execution

MichaIng committed:
v6.35

+ DietPi-PREP | Remove various base files and all home directories (including /root) in one step and restore via base-files package reinstall
+ DietPi-PREP | Exclude another Armbian skeleton dir content which doubles with DietPi htoprc
+ DietPi-PREP | On post-v6.28 sytems, when a DietPi base system is detected, stop dietpi-preboot explicitly to have dietpi-boot and dietpi-postboot stopped with it as requisite

MichaIng committed:
v6.35

+ DietPi-PREP | Try to mute a perl locale error by exporting safe locale variables before reconfiguring the locale package

MichaIng committed:
v6.35

+ DietPi-PREP | Add dietpi.com SSH host key after cleaning /root home. Fix base-files reinstall not re-creating missing config files and call postinst manually, since on reinstall with a shipped arg $2 existing version, creating further base configs and dirs is skipped.

MichaIng committed:
v6.34

+ DietPi-Patch | Fix broken Sparky SBC kernel patch

MichaIng committed:
v6.34

+ DietPi-Patch | Fix another Sparky SBC patch error

MichaIng committed:
v6.35 (#4073)

DietPi-Patch | Merge some Sparky SBC related fixes back to master

MichaIng committed:
v6.35

+ DietPi-Software | Medusa: Disable on Stretch, since Medusa now requires Python 3 > 3.5, which is not available on Debian Stretch
+ DietPi-Software | Medusa: Prevent unlimited service restarts on failure: Permit up to three restarts in 10 minutes

MichaIng committed:
v6.35

+ CHANGELOG | Medusa: This software option has been disabled on Stretch systems

Joulinar committed:
v6.35 (#4039)

+ DietPi-Software | PaperMC: Add memory parameter to the service to allow successful initial start of the service on low memory devices.
+ DietPi-Software | PaperMC: Ensure a memory setting of min 1.5 GiB (phys + swap). If needed, create a swap file
+ DietPi-Software | PaperMC: Show an information about long start time and swap use on software selection
+ DietPi-Software | PaperMC: Added new download URL to obtain latest PaperMC version
+ DietPi-Software | Create_Config: Allow to print service journal output to console for possible time consuming startups, so that users can follow the process. Additionally stop waiting if the service is not actively running anymore, e.g. due to a crash or other unexpected stop.
+ DietPi-Software | PaperMC: Print journal output of possible long taking service startup to console
+ DietPi-Software | PaperMC: Do not ask for EULA when it has been accepted before already

Joulinar committed:
v6.35

DietPi-Software | WiringPi: add a check for availability of /usr/local/bin

MichaIng committed:
v6.35

+ CHANGELOG | WiringPi: Resolved an issue where the install failed, if the /usr/local/bin directory was not present

nonnorm committed:
v6.35 (#3989)

+ DietPi-Software | Box86: An i386 wrapper for ARMv7 systems is now available for install. Thanks to binfmt, i386 binaries are executed automatically through Box86, which is able to either wrap native ARMv7 system libraries to work with the i386 binary or find and emulate i386 libraries, installed on the system. Note that it is not able to run x86_64 binaries, nor load such system libaries.
+ DietPi-Software | Steam: It can now be installed on ARMv7 systems, which will pull in Box86 as dependency then. Note that features and stability are limited and not all games will run. You'll be able to read more about it in our documentation (TBD): https://dietpi.com/docs/

C0D3-M4513R committed:
v6.35 (#4015)

+ DietPi-Software | Consequently use $G_GITBRANCH instead of hardcoded "MichaIng" Git owner
+ DietPi-Software | Add desktop configs system-wide instead of for root's desktop only
+ DietPi-Software | Desktop shortcuts are now created as follows: On first desktop install, a system-wide desktop autostart entry is created that links/copies user-level configs and desktop entries into the users home dir, once it logs in the first time. The autostart entry then disables itseld. This way only users which are actually used for desktop sessions get related configs and entries, and only a single time. Desktop shortcuts for later installed software titles are added only to users, which did already log into a desktop, hence have the autostart entry disabled. For now users, those are added by the autostart entry as well.
+ DietPi-Software | Further implement G_EXEC error handler where applicable and minor coding
+ DietPi-Software | DietPi-Desktop_setup: Replace /root in bookmarks with actual users home
+ DietPi-Software | MATE: Enable audio (playback and settings panel) by installing the mate-media package
+ DietPi-Software | MATE: Do not add a Caja desktop icon, since by default a user home icon is present already
+ DietPi-Software | Desktops: Simplify xcompmgr install, which is only used/compatible with LXDE and LXQt. All other desktops have one integrated so that either the integrated one or xcompmgr fails. To be failsafe on Xfce, actively remove the autostart entry as it breaks the internal one, leading to ugly results.
+ DietPi-Software | X server: Do not install xterm by default. Each desktop has an own terminal emulator which looks and works much better. Especially when starting terminal applications from desktop entries, xterm overrides the desktops emulator, but has limited colour and fonts set, which does not look good. If there is a non-desktop X application that can make use of a terminal emulator without shipping one, we can re-add it to the related X client install.
+ DietPi-Software | GNUstep: Add xterm as dependency as the GPUstep "Terminal" cannot launch scripts. Remove "x-window-system-core" => "xorg" and "libc-dbg" => "libc6-dbg" dependencies which are not required to launch any of the pre-installed desktop applications.

MichaIng committed:
v6.35

+ DietPi-Software | Steam: Do no add Steam desktop shortcut for root user anymore, which is not done on first login: https://github.com/MichaIng/DietPi/pull/4015

MichaIng committed:
v6.35

+ DietPi-PREP | Actively enable weekly TRIM on all mounted drives that support it
+ DietPi-PREP | Use error handler and be more transparent about disabling e2scrub and apt-daily
+ DietPi-PREP | Remove doubled "base-files" argument

MichaIng committed:
v6.35

+ DietPi-Patch | RPi: Offer to remove the automatically added MS VS Code repository

MichaIng committed:
v6.35

+ DietPi-PREP | RPi: Remove MS VS Code repository by default

MichaIng committed:
v6.35 (#4084)

+ DietPi-Survey_report | Add Box86 and Docker Compose recognition
+ CHANGELOG | Add Docker Compose, Box86 and Steam on ARM software implementation notes and PaperMC fixes
+ README | Add Docker Compose, Box86 and Steam repo/community URLs

MichaIng committed:
v6.35 (#4086)

+ DietPi-Software | Migrate documentation URLs shown by dietpi-software to our new documentation site and add a few missing URLs: https://dietpi.com/docs/
+ DietPi-Software | Xfce Power Manager: Rename it so that it matches how the name is shown within its GUI
+ DietPi-Software | X.Org X Server: Rename it to include the actual implementation we install
+ DietPi-Software | Rephrase or extend a few software descriptions, which were a bit short or nor precise

StephanStS committed:
Comment changed from dietpi.com/phpBB to dietpi.com/docs

Old: https://dietpi.com/phpbb/viewtopic.php?p=34#p34  
New: https://dietpi.com/docs/dietpi_tools/#miscellaneous (tab 'DietPi Survey')

StephanStS committed:
Update dietpi.txt

StephanStS committed:
Update dietpi-survey

StephanStS committed:
Merge pull request #4090 from MichaIng/StephanStS-patch-1

Comment changed from dietpi.com/phpBB to dietpi.com/docs

MichaIng committed:
v6.35

+ DietPi-Patch | Only attempt to remove VS Code repository if we're on RPi and the file exists

MichaIng committed:
v6.35 (#4091)

+ DietPi-Software | Add mjpg-streamer install option
+ DietPi-Survey_report | Add v6.35 mjpg-streamer support, migrate/apply new combined Odroid C4/HC4 SBC name and new X.Org X server naming
+ DietPi-Software | mjpg-streamer: Use RPi camera module input plugin by default, if it is enabled
+ DietPi-Software | mjpg-streamer: Inform users that the stream by default is unprotected and that security measures should hence be applied, e.g. allowing public access, if required, only via VPN.
+ DietPi-Software | mjpg-streamer: Disable camera stream commands by default. I'm currently not sure what can be done with it, but it requires some manual setup anyway.
+ DietPi-Software | mjpg-streamer: Since IP binding is perfectly possible and reasonable in certain cases, the service must not start before network interfaces have been configured.
+ DietPi-Software | mjpg-streamer: Apply additional service hardenings by default, which do not break the default camera input HTTP output use-case. The online documentation needs to contain info about how to allow R/W access to file system paths, e.g. for use with file input and/or output plugins.
+ DietPi-Software | mjpg-streamer: If it is installed along with OctoPrint, configure OctoPrint to use mjpg-streamer for it's webcam features. Also install FFmpeg, required it's time-lapse feature.

MichaIng committed:
v6.35

+ DietPi-Software | Unbound: Create monthly cron job to keep root hints updated

MichaIng committed:
v6.35

+ CHANGELOG | Unbound: A monthly cron job is now created to keep the root hints file updated

MichaIng committed:
v6.35

+ DietPi-Patch | Gracefully handle missing /mnt/dietpi_userdata: If it is required to apply migration steps, pre-create it, else apply changes to it conditionally. Do not create config file/dir backups in /mnt/dietpi_userdata, but instead just beside the original file/dir with .bak/_bak suffix.
+ DietPi-Patch | Apply G_EXEC further throughout the script

nonnorm committed:
v6.35 (#4094)

+ DietPi-Backup | Do not apply and check the file system when "Cancel" was selected in any of the target location selection menus
+ DietPi-Backup | Coding: Instead of creating a variable to check back indirectly, make Check_Supported_Directory_Location return it's result via exit code and check it back directly.
+ DietPi-Backup | Coding: Instead of using a large indented if-then-else block in Backup and Restore functions for the Check_Supported_Directory_Location result, return to menu directly, when the file system is not supported. For readability, other check should be coded in the same way, but it's enough for today :).
+ DietPi-Backup | When searching for previous backups as target location and none could be found, skip the supported file system check and do not touch the target directory variable
+ DietPi-Backup | Pre-select "Help" main menu entry initially
+ DietPi-Backup | Since the exit menu verification requires interactive confirmation, show main menu only when interactivity is guaranteed. Else it is possible to run into an endless loop via "G_INTERACTIVE=0 dietpi-backup" or when calling the script without "1"/"-1" arguments from cron job, systemd unit or else without valid terminal at STDIN.
+ DietPi-Backup | Do not check for mounted dietpi_userdata during supported target location check. It cannot be used as target location anyway, regardless of mount state, and if it is a mount/symlink, it is currently excluded from backup anyway. If/when we change this behaviour, to (optionally) include external drives/userdata into the system backup, mount states in general should be checked, and in case unmounted drives need to be excluded for the backup/restore run to neither remove an intact backup of a currently unmounted drive, nor restore an external drive backup to the root file system.

MichaIng committed:
v6.35

+ DietPi-Patch | RPi: Remove additional APT preferences installed now together with the raspberrypi-sys-mods package, when user chooses to remove the VS Code repo, as it might cause conflicts with other repositories and at best is obsolete then.

MichaIng committed:
v6.35

+ DietPi-PREP | RPi: Remove additional APT preferences installed now together with the raspberrypi-sys-mods package, as it might cause conflicts with other repositories and at best is obsolete then.

Joulinar committed:
v6.35 (#4096)

+ DietPi-Software | Domoticz: Move scripts directory to data dir: https://dietpi.com/phpbb/viewtopic.php?t=8627
+ DietPi-Software | Add a missing config banner + a few G_EXEC and missing service config dir removals

MichaIng committed:
v6.35

+ CHANGELOG | Domoticz: Resolved an issue where saving custom scripts and starting with a template did not work

MichaIng committed:
DietPi-Software | Several enhancements (#4095)

+ DietPi-Software | Remove mandatory reboot after installs, instead manually start up installed services, not controlled by DietPi-Services
+ DietPi-Software | GNUstep: Fix spelling
+ DietPi-Software | wiringPi / phpSysInfo: Gracefully handle reinstalls, i.e. when the install dir exists already, preserving config files
+ DietPi-Software | Do not upgrade all package on install. The daily APT update check helps now to inform users about available updates and the minimum required package versions we need to count on are ensured via dietpi-update. Since the package cache update function prints sufficient information, remove the related large "Banner_Apt_Update" banner.
+ DietPi-Software | Do not pre-create /var/lib/dietpi/dietpi-software/services directly, which is not used anymore since several DietPi updates
+ DietPi-Software | Do not re-apply dietpi-autostart non-interactively on installs. /boot/dietpi/.dietpi-autostart_index is not touched during installs, hence the step applies no change. On first run installs, the related dietpi.txt setting is applied. On interactive installs of software with autostart support, users are offered to interactively change the autostart option, which applies for next boot and includes some information and choices like the autostart user.
+ DietPi-Software | Remove the "Banner_Configs" banner. We have no "Banner_Installs" either, as every installed/configured software title prints its own banner already.
+ DietPi-Software | Apply G_EXEC further throughout the script where applicable
+ DietPi-Software | Raspimjpeg: Turn service into a bourne shell script, bash is overkill here
+ DietPi-Software | Raspimjpeg: Fix location info inside of shell script
+ DietPi-Patch | Remove obsolete /var/lib/dietpi/dietpi-software/services directory which was still pre-created on every dietpi-software install while not actually used since a few versions
+ DietPi-Software | LXDE/LXQt: Fix xcompmgr.service download, which is stored as X server config
+ DietPi-Software | LXDE: Replace single colour panel background image with solid background colour
+ DietPi-Software | LXDE: Slightly harden pcmanfm configuration by dietpi-desktop_setup
+ DietPi-Software | Desktops: Call users home dir in file manager bookmarks "Home"
+ DietPi-Software | OpenTyrian: Fix dependency typo, which was luckily accepted and correctly resolved by APT as regex match. Disable install on x86_64 system, as the Debian package has a very different file structure and required data build via dedicated "game-data-packager" package. Also disable for 64-bit RPi where functionality would need to be verified first, likely a 64-bit binary would need to be compiled.
+ DietPi-Set_hardware | soundcard: Do not error-handle "alsactl init", as it returns an error code even on success. Also do not use "-g" option (ignore if no sound card was found), which is valid for store and restore commands.

MichaIng committed:
v6.35

+ DietPi-Software | Unbound: Since IP binding might be used, start service after network interfaces have been configured, not when they just start to be configured

MichaIng committed:
v6.35

+ DietPi-CloudShell | Fix and simplify obtaining Pi-hole stats, by asking FTL via it's TCP port API directly, instead of querying any logs or database files manually. This matches the method how Pi-hole's Chronometer and PADD are doing it: https://github.com/pi-hole/pi-hole/blob/development/advanced/Scripts/chronometer.sh, https://docs.pi-hole.net/ftldns/telnet-api/
+ DietPi-CloudShell | Massively simplify uptime estimation: Instead of looping through each minute, then each hour, then each day, calculate hours and days in a single step
+ DietPi-CloudShell | When obtaining drive data hangs, do not killall df executions but be more specific by storing the PID and killing the exact process only
+ DietPi-CloudShell | Do not initialise variables with "0" when they are not aimed to be integers

MichaIng committed:
v6.35

+ DietPi-CloudShell | Pi-hole scene: /dev/tcp/127.0.0.1/$port does not exist in the file system, but as possible file descriptor only, hence remove the existence check
+ DietPi-CloudShell | Pi-hole scene: The FTL API allows only one output command per time, hence get last blocked domain first, without closing the connection, then get the other stats and close the connection.
+ DietPi-CloudShell | Pi-hole scene: Fix syntax to parse one line per loop via read

MichaIng committed:
v6.35

+ DietPi-CloudShell | Pi-hole scene: Prevent division with zero, when no queries have been processed yet
+ DietPi-CloudShell | Pi-hole scene: FTL cannot handle a second command when the first commands output has not fully been read. Follow Chronometer/PADD in closing the connection after the first command and re-establishing it before sending the second command.
+ DietPi-CloudShell | Pi-hole scene: Show function exit code when FTL connection failed, for debugging purpose
+ DietPi-CloudShell | Pi-hole scene: Fix detection of Pi-hole install by checking for pihole-FTL command

MichaIng committed:
v6.35

+ DietPi-CloudShell | Show more precise error message when Pi-hole FTL connection fails

MichaIng committed:
v6.35 (#4100)

+ DietPi-CloudShell | Fix and simplify obtaining Pi-hole stats, by asking FTL via it's TCP port API directly, instead of querying any logs or database files manually. This matches the method how Pi-hole's Chronometer and PADD are doing it: https://github.com/pi-hole/pi-hole/blob/development/advanced/Scripts/chronometer.sh, https://docs.pi-hole.net/ftldns/telnet-api/
+ DietPi-CloudShell | When obtaining FTL data fails, show a meaningful error message based on the stage which failed
+ DietPi-CloudShell | Massively simplify uptime estimation: Instead of looping through each minute, then each hour, then each day, calculate hours and days in a single step
+ DietPi-CloudShell | When obtaining drive data hangs, do not killall df executions but be more specific by storing the PID and killing the exact process only
+ DietPi-CloudShell | Do not initialise variables with "0" when they are not aimed to be integers

MichaIng committed:
v6.35

+ DietPi-Update | Add new remote version file for updates and automatic branch migration from next version on

MichaIng committed:
v6.35

+ DietPi-Pre-patches | Create new pre-patches file in new separate update directory outside of DietPi core scripts directory

MichaIng committed:
v6.35

+ DietPi-Update | Apply automatic branch migration and new remote version + pre-patches URLs

MichaIng committed:
v6.35

+ DietPi-Update | Remove https://dietpi.com/ update server. Access to the GitHub repository is required anyway to download the dietpi.txt and for various software configs, when being patched or reinstalled, so it's even better to exit in the very first place instead of running into failing connections at a later stage.

MichaIng committed:
v6.35

+ DietPi-Update | Remove obsolete CHANGELOG_DOWNLOADED declaration
+ DietPi-Update | Check desktop session via $DISPLAY variable to detect parent X session, like we do everywhere else. lxsession detects LXDE desktop sessions only.

MichaIng committed:
v6.35

+ DietPi-Update | Both, pre-patches and incremental patches for v6 still need to be applied since DietPi-Update restarts are done before incremental patches have been applied and pre-v6.17 system didn't have pre-patches applied either.

MichaIng committed:
v6.35

+ DietPi-Update | Create DietPi v7 incremental patch file

MichaIng committed:
v6.35

+ DietPi-Patch | This is only called by DietPi v6 systems, so restart whenever it is not the second instance already

MichaIng committed:
v6.35 (#4106)

+ DietPi-Software | Fix download GitHub URL/version detection for ruTorrent, Koel and Bitwarden_RS: mawk does not support "[[:blank:]]" syntax, hence the fallback URL was used in those cases.
+ DietPi-Software | Align automated GitHub download URL/version detection for all installs. Take into account that mawk is not able to handle "[[:blank:]]" syntax and variables, hence if this is required, "grep | cut" pipe needs to be used. Use "curl -sSfL" in all cases so that curl errors and server error responses are shown, for debugging purpose. mawk and grep are then not allowed to exit on first match, since this causes a curl 23 exit code and error message about a broken pipe before the response could have been written completely, when the match was not within the last buffer chunk. To compensate this, assure that a single match exists by being more specific about the expected regex, assure that a fallback is present and effective in every case.
**Once Stretch support is dropped, we could use `CURLOPT_BUFFERSIZE=524288` (introduced with curl v7.53 vs Stretch curl v7.52.1) to maximise the curl buffer chunks so that a GitHub API page fully fits inside and parsing can stop at first match. But being explicit about the URL has it's own benefits, in case of multiple matches we might want to review the two options and pick one explicitly.**
+ DietPi-Software | Whenever a download fallback URL or version string is used, print an error message, inform about the fallback but ask user to do a report as obviously we need to update the URL or regex.
+ DietPi-Software | "curl -IL" does not suffer from failing GitHub download URL checks, hence the workaround to check the API page only can be removed. Testet on all distros from Stretch to Bullseye.
+ DietPi-Software | Use G_EXEC more widely throughout the script, where applicable

MichaIng committed:
v6.35

+ CHANGELOG | DietPi-Update: A new update check system has been implemented, which checks for a minimum required Debian and DietPi version and migrates systems with either too old Debian version or too old DietPi version to a different Git branch automatically.

MichaIng committed:
v6.35 (#4107)

Update "update" branch

MichaIng committed:
v6.35

+ DietPi-Update | Fix incremental patch file path

MichaIng committed:
v6.35

+ DietPi-Update | Apply execute permissions to patch file

MichaIng committed:
v6.35

+ DietPi-Patch | Loop until hardcoded v6.35 since this the last v6 subversion string.

MichaIng committed:
v6.35

+ DietPi-Patch | Failsafe: If for some reason the v6 subversion is >35, exit loop as well

MichaIng committed:
v6.35

+ DietPi-Survey_report | Add DietPi v7.0 support, matching DietPi v6.35 transition version string

MichaIng committed:
v6.35

+ DietPi-Survey_report | Satisfy shellcheck

MichaIng committed:
v6.35

+ DietPi-Survey_report | Do not trust negative CPU temperatures (for now) and time or bandwidth results of zero (usually an error during benchmark) to reduce benchmark results distortion.
+ DietPi-Survey_report | Eliminate the need for a hardcoded first min/max benchmark value by using the first individual result instead, being simply correct.

MichaIng committed:
v6.35

+ DietPi-Update | Move APT auto-removal and systemd unit reload after incremental patches from patch file to main script, since we currently handle it as mandatory step anyway.

MichaIng committed:
v6.35

+ DietPi-Patch | APT auto-removal and systemd unit reload is now done within dietpi-update parent script.

MichaIng committed:
v6.35

+ DietPi-Update | After three years, remove the DietPi pre-v6 version file. This means that these DietPi systems won't see a "new image available" message, indicating that a new image needs to be flashed, but the update check will fail instead, which practically the same can be derived from.

MichaIng committed:
v6.35

+ DietPi-Update | "DietPi-$GITBRANCH_TARGET/dietpi/server_version" does not exist anymore on v7 branches

MichaIng committed:
v6.35

+ CHANGELOG | Typo

MichaIng committed:
v6.35

+ DietPi-Cron_daily | Remove  MOTD file and re-create it only when the download succeeds

MichaIng committed:
v6.35

+ DietPi-Banner | Make the menu the default input argument, so it can be made a console alias reasonably. This means that without input argument, the banner won't be printed anymore, but "0" is required explicitly.

MichaIng committed:
v6.35

+ DietPi-Banner | Allow 3 seconds for network connections
+ DietPi-Banner | Create MOTD file only in case of download success, so we don't need to assume a failure when it's empty, which simplifies the checks. In turn we can leave it empty intentionally to have the banner line skipped.

MichaIng committed:
v6.35

+ DietPi-Bash | Update dietpi-banner alias: Without an input argument, it opens the menu now, so using "dietpi-banner 0" or "dietpi-banner 1" can now be used to print the banner.

MichaIng committed:
v6.35

+ DietPi-Bash | Fix wrong OpenTyrian alias

MichaIng committed:
v6.35

+ CHANGELOG | DietPi-Banner: Running the script without input argument will now open the menu instead of printing the banner. The console alias has been adjusted accordingly, so that it is now possible to run "dietpi-banner 0" and "dietpi-banner 1" from console to have full and short banner versions printed. The "dietpi-banner" console command remains opening the menu.

MichaIng committed:
v7.0

+ Use RC version "-1" before releasing the first beta, as before

MichaIng committed:
v7.0

+ DietPi-Globals | Default core up

MichaIng committed:
v7.0 (#4103)

+ DietPi-Update | Add new remote version file for updates and automatic branch migration from next version on
+ DietPi-Pre-patches | Create new pre-patches file in new separate update directory outside of DietPi core scripts directory
+ DietPi-Update | Apply automatic branch migration and new remote version + pre-patches URLs
+ DietPi-Update | Remove https://dietpi.com/ update server. Access to the GitHub repository is required anyway to download the dietpi.txt and for various software configs, when being patched or reinstalled, so it's even better to exit in the very first place instead of running into failing connections at a later stage.
+ DietPi-Update | Check desktop session via $DISPLAY variable to detect parent X session, like we do everywhere else. lxsession detects LXDE desktop sessions only.
+ DietPi-Survey_report | Add DietPi v7.0 support, matching DietPi v6.35 transition version string
+ DietPi-Update | Move APT auto-removal and systemd unit reload after incremental patches from patch file to main script, since we currently handle it as mandatory step anyway.
+ DietPi-Update | After three years, remove the DietPi pre-v6 version file. This means that these DietPi systems won't see a "new image available" message, indicating that a new image needs to be flashed, but the update check will fail instead, which practically the same can be derived from.

Joulinar reviewed (dismissed):
Let's give it a try and go for first beta 😃

MichaIng committed:
v7.0

+ DietPi-Drive_Manager | Change comment in fstab to indicate that dietpi-drive_manager is not a must, but an offer, and that it overwrites and re-creates physical drive mount entries on use
+ DietPi-Drive_Manager | Append df/blkid/lsblk outputs to detection loops directly: It creates a temporary file as well, but assures a unique safe path and auto-removal in a native way, faster as well.
+ DietPi-Drive_Manager | Add /dev/root workaround right into df output pipe to let mawk handle the possibly left duplicate

MichaIng committed:
v7.0

+ DietPi-Software | Nextcloud v21: Redirect webfinger and nodeinfo requests to Nextcloud endpoint

MichaIng committed:
v7.0

+ DietPi-Software | Nextcloud v21: Redirect webfinger and nodeinfo requests to Nextcloud endpoint to mute admin panel warnings

MichaIng committed:
v7.0

+ DietPi-Software | Nextcloud: Update Nginx config based on enhanced config from Nextcloud docs. Redirect webfinger and nodeinfo requests to Nextcloud endpoint to mute admin panel warnings and remove obsolete OCM/OCS provider redirects

MichaIng committed:
v7.0

+ DietPi-Software | Nextcloud: Remove obsolete OCM/OCS provider redirects

MichaIng committed:
v7.0

+ DietPi-Software | Nextcloud: Update Lighttpd config based on Nginx config updates. Test applying security headers for whole location and pass the "modHeadersAvailable" to PHP to avoid doubled headers.

MichaIng committed:
v7.0

+ DietPi-Software | Nextcloud: Revert Lighttpd config change partly, since multiple matching $HTTP["url"] section do not stack, but the last/best match overrides prior/weaker matches. To avoid massively doubled code, merge all into a single static resource block. Long webfonts cache time should not be an issue, also leave .png and .map files inside.

Joulinar commented:
just an observation as I was testing the backup feature. I noticed that `vscode.list` did not get removed during initial update, even if I select `remove it`

MichaIng committed:
v7.0

+ DietPi-Patch | Fix doubled negation on VS Code repo removal

MichaIng commented:
Good spot! I had a doubled negation inside, so "Keep it" would have removed it 😅: https://github.com/MichaIng/DietPi/commit/dffd6427be3643be16293a22875853f035621bd0
Note that, if a system is on 7.X version already, that patch file is not executed anymore, but only the new (currently empty) one.

I think for us it's okay, but we need to keep in mind when changing the patches during beta phase, we need to move the affected patches to have them re-applied when upgrading von 7.0.0 to 7.0.1.

MichaIng committed:
v7.0

+ DietPi-Software | Python 3 pip: get-pip.py does not support Python 3.5 anymore, but a dedicated backwards-compatible script is luckily available

MichaIng committed:
v7.0

+ DietPi-Backup | Add support for XFS and ZFS and update error message to show all supported file systems (excluding NFS, which would require more explanation)

MichaIng committed:
v7.0

+ CHANGELOG | DietPi-Backup: Support for XFS and ZFS target file system types has been added, which fully support the required symlink and POSIX permissions capabilities.

MichaIng committed:
v7.0

+ DietPi-Backup | Clear the PATH cache after restoring a backup, in case command paths have changed, e.g. due to /usr/bin /bin merge
+ DietPi-Backup | Syntax

MichaIng committed:
v7.0

+ DietPi-Software | Tiny

MichaIng committed:
v7.0

+ DietPi-Software | Do not stop (and restart) services on software uninstalls. The uninstalled titles' services are stopped as first step of their individual uninstall block. Since uninstalls don't require much memory or CPU resources, other services can stay active without risk. If dependencies are uninstalled, dependants may crash, but previously their restart would have failed as well.
+ DietPi-Software | When the "uninstall" input argument is used, do not run the uninstall function when no software title has been actually marked for uninstall, i.e. none of the input IDs is currently installed
+ DietPi-Software | Handle input arguments in a safer way

MichaIng committed:
v7.0

+ CHANGELOG | DietPi-Software: Uninstalls do not stop other services anymore

MichaIng committed:
v7.0

+ DietPi-Software | LXQt: Update configuration, fix visual panel issues, remove obsolete files, cleanup

MichaIng committed:
v7.0

+ DietPi-Software | LXQt comes with xfwm4 window manager as well, with internal composite manager that conflicts with xcompmgr, like Xfce does. So do not install xcompmgr with LXQt and assure that it's disabled otherwise.
+ DietPi-Software | X server: Do not install xserver-xorg package, which includes all display drivers with are mostly not required or conflict or override the often well working modesetting fallback, at least better than VESA, which would be prioritised. For PCs we have the Intel/Nvidia/AMD selection already, which now makes even sense for AMD GPUs. Outstanding tests is performance on RPi desktops vs fbdev driver or fbturbo, available on RPi repo, and VMware display driver for VMs (as well VirtualBox can emulate the VMware GPU).

MichaIng committed:
v7.0

+ DietPi-Software | X.Org X server: Install VMware GPU driver for all VMs
+ DietPi-Software | Home Assistant: Bump Python to v3.8.7. 3.8 is still the target version. 3.9 works mostly but a few related issues are present on GitHub, hence we wait for another release before switching to Python 3.9.

MichaIng committed:
v7.0

+ CHANGELOG | XQt: Resolved visual issues with our default configuration of Debian Buster
